File: api_nestjs_distributed_order_platform/src/application/port/stock.repository.ts
import { StockEntity } from "src/domain/entities/stock.entity";
import { Transaction } from "sequelize";

export interface StockRepositoryInterface {
    create(stock: StockEntity): Promise<StockEntity>;
    findByProductId(productId: string): Promise<StockEntity | null>;
    findAll(): Promise<StockEntity[]>;
    update(stock: StockEntity, expectedVersion?: number): Promise<boolean>;
    delete(productId: string): Promise<void>;
    reserve(
        productId: string,
        quantity: number,
        transaction?: Transaction,
    ): Promise<StockEntity>;
    release(
        productId: string,
        quantity: number,
        transaction?: Transaction,
    ): Promise<StockEntity>;
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/application/port/StockRepository.java
package com.project.order.application.port;

import com.project.order.domain.model.Stock;

import java.util.List;
import java.util.Optional;

public interface StockRepository {
    Stock create(Stock stock);
    Optional<Stock> findById(String productId);
    List<Stock> findAll();
    Stock update(Stock stock);
    void delete(String productId);
}
